tests/driver/runner_test.py

```diff
@@ -2779,36 +2914,56 @@ async def test_create_snapshot_wait_for_completion(self, es):
         params = {
             "repository": "backups",
             "snapshot": "snapshot-001",
-            "body": {
-                "indices": "logs-*"
-            },
-            "wait-for-completion": True,
-            "request-params": {
-                "request_timeout": 7200
-            }
         }
 
-        r = runner.CreateSnapshot()
+        r = runner.WaitForSnapshotCreate()
         result = await r(es, params)
 
         self.assertDictEqual({
             "weight": 9399505,
             "unit": "byte",
             "success": True,
-            "service_time": 0.2,
-            "time_period": 0.2
+            "duration": 200,
+            "file_count": 70,
+            "throughput": 46997525.0,
+            "start_time_millis": 1591776481060,
+            "stop_time_millis": 1591776481060 + 200
         }, result)
 
-        es.snapshot.create.assert_called_once_with(repository="backups",
+        es.snapshot.status.assert_called_once_with(repository="backups",
                                                    snapshot="snapshot-001",
-                                                   body={
-                                                       "indices": "logs-*"
-                                                   },
-                                                   params={"request_timeout": 7200},
-                                                   wait_for_completion=True)
+                                                   ignore_unavailable=True)
 
-        es.snapshot.status.assert_called_once_with(repository="backups",
-                                                   snapshot="snapshot-001")
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_wait_for_snapshot_create_failure(self, es):
+        snapshot_status = {
+            "snapshots": [
+                {
+                    "snapshot": "snapshot-001",
+                    "repository": "backups",
+                    "state": "FAILED",
+                    "include_global_state": False
+                }
+            ]
+        }
+        es.snapshot.status.return_value = as_future(snapshot_status)
+
+        params = {
+            "repository": "backups",
+            "snapshot": "snapshot-001",
+        }
+
+        r = runner.WaitForSnapshotCreate()
+
+        logger = logging.getLogger("esrally.driver.runner")
```

Comment:
how about we just use the runner's logger with `logger = r.logger` (you can then probably just inline it in the `with` statement below? This would also make it more refactoring safe.

Reply:
Addressed in 32ee4abf6a15d4158992c9843958b51e9a766cb5